packages/react-components/react-dialog/src/components/DialogTitle/useDialogTitle.tsx

```diff
@@ -17,21 +18,21 @@ import { useDialogTitleInternalStyles } from './useDialogTitleStyles.styles';
  * @param ref - reference to root HTMLElement of DialogTitle
  */
 export const useDialogTitle_unstable = (props: DialogTitleProps, ref: React.Ref<HTMLElement>): DialogTitleState => {
-  const { as, action } = props;
+  const { as = defaultComponentType, action } = props;
   const modalType = useDialogContext_unstable(ctx => ctx.modalType);
   const internalStyles = useDialogTitleInternalStyles();
 
   return {
-    components: {
-      root: 'h2',
-      action: 'div',
-    },
-    root: getNativeElementProps(as ?? 'h2', {
-      ref,
-      id: useDialogContext_unstable(ctx => ctx.dialogTitleId),
-      ...props,
-    }),
-    action: resolveShorthand(action, {
+    root: slot<DialogTitleProps>(
+      getNativeElementProps(as, {
+        ref,
+        id: useDialogContext_unstable(ctx => ctx.dialogTitleId),
+        ...props,
+      }),
+      { componentType: defaultComponentType, required: true },
```

Comment:
```suggestion
      { componentType: as, required: true },
```

Why not `as`?

Reply:
It actually should be `as` in that case! otherwise there's no way to alter `root` base type, since `getNativeElementProps` will do us the favor of removing the `as` property 🥲